### ISSUE TITLE
Processing of OpenAPI Specs with OData paths results in appearance of '?'

### ISSUE THREAD

**Issue by robot-accomplice** [mikefarah/yq#1033]
When executing an eval statement against a target document, the lines described below are minutely transformed to include an unallowable '?' character.

Version of yq: 4.16.1/4.12.0
Operating system: mac/linux
Installed via: homebrew/git clone

**Input Yaml**
(apologies about indenting, the markup isn't playing nice)
data1.yml:
```yaml
          $ref: '#/components/responses/error'
 
 '/DocumentInfoRecordAttch(DocumentInfoRecordDocType=''{DocumentInfoRecordDocType}'',DocumentInfoRecordDocNumber=''{DocumentInfoRecordDocNumber}'',DocumentInfoRecordDocVersion=''{DocumentInfoRecordDocVersion}'',DocumentInfoRecordDocPart=''{DocumentInfoRecordDocPart}'')/DocumentInfoRecordToAttachmentNavigation':
    parameters:
      - name: DocumentInfoRecordDocType
```

**Command**
The command you ran:
```
unzip -p ${bundle} docs/openapi.yaml | yq e '.x-rlg-api = env(api)' - > $tmp_file
```

**Actual behavior**

```yaml
          $ref: '#/components/responses/error'
  ? "/DocumentInfoRecordAttch(DocumentInfoRecordDocType=''{DocumentInfoRecordDocType}'',DocumentInfoRecordDocNumber=''{DocumentInfoRecordDocNumber}'',DocumentInfoRecordDocVersion=''{DocumentInfoRecordDocVersion}'',DocumentInfoRecordDocPart=''{DocumentInfoRecordDocPart}'')/DocumentInfoRecordToAttachmentNavigation"
  : parameters:
```

**Expected behavior**

```yaml
          $ref: '#/components/responses/error'
 '/DocumentInfoRecordAttch(DocumentInfoRecordDocType=''{DocumentInfoRecordDocType}'',DocumentInfoRecordDocNumber=''{DocumentInfoRecordDocNumber}'',DocumentInfoRecordDocVersion=''{DocumentInfoRecordDocVersion}'',DocumentInfoRecordDocPart=''{DocumentInfoRecordDocPart}'')/DocumentInfoRecordToAttachmentNavigation':
    parameters:
```

I'm happy to supply any missing information.

**Comment by mikefarah** [comment]
The data1.yaml file isn't valid yaml. Please provide a short, valid yaml document as input and an example 'api' env so I can replicate the issue.

**Comment by robot-accomplice** [comment]
Apologies, I went on vacation immediately after posting this and have not had time to go back and produce a sanitized version of the original file.  Based on experimentation and the fact that this issue also manifests in VSCode, I suspect the source of the problem (no pun intended) is an upstream dependency.  I have validated the offending text and it *is* in fact valid YAML -- even if the syntax is weird and confusing.

I will work on removing any identifying information from the OpenAPI Spec and attach it the issue and/or open a new issue.
